packages/trader/src/Modules/SmartChart/Components/Markers/accumulators-profit-loss-text.tsx

```diff
@@ -16,24 +27,29 @@ const AccumulatorsProfitLossText = ({
     currency,
     className = 'sc-accumulators-profit-loss-text',
     profit,
-}) => {
+}: TAccumulatorsProfitLossText) => {
     const [is_fading_in, setIsFadingIn] = React.useState(false);
     const [is_sliding, setIsSliding] = React.useState(false);
-    const prev_profit = React.useRef(profit);
+    const prev_profit = React.useRef<number>(profit);
     const prev_profit_tenth = +prev_profit.current?.toFixed(2).split('.')[1][0];
     const [current_profit_tenth, setCurrentProfitTenth] = React.useState(prev_profit_tenth);
-    const profit_tenth_ref = React.useRef();
-    const interval_id_ref = React.useRef(null);
-    const fading_in_timeout_id = React.useRef();
-    const sliding_timeout_id = React.useRef();
+    const profit_tenth_ref = React.useRef<number>(0);
+    const interval_id_ref = React.useRef<ReturnType<typeof setInterval>>();
+    const fading_in_timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    const sliding_timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
     const profit_portions_array = profit.toFixed(2).split('.');
     const profit_whole_number = +profit_portions_array[0];
     const profit_tenth = +profit_portions_array[1][0];
     const profit_hundredths = +profit_portions_array[1].slice(1);
     const won = profit >= 0;
     const sign = profit > 0 ? '+' : '';
 
-    const runThroughTenthDigit = (action, interval_ms, start, end) => {
+    const runThroughTenthDigit = (
+        action: TActions[keyof TActions],
```

Comment:
```suggestion
        action: typeof keyof ACTIONS,
```
will work together with https://github.com/binary-com/deriv-app/pull/8257/files#r1168463465

Reply:
```suggestion
        action: keyof typeof ACTIONS,
```
my bad😊 I switched `typeof` & `keyof` places

Reply:
this works - `typeof ACTIONS[keyof typeof ACTIONS]`

`keyof typeof ACTIONS` refer to INC, DEC, ADD10

where as `typeof ACTIONS[keyof typeof ACTIONS]` refer to 'increment', 'decrement' and 'add10'